Refactor: Use a unified function to output charge density and potential in real space.

sunliang98 opened the pull request:
1. Unify the format of charge density, effective potential and electrostatic potential into `cube`. In detail, rename `Module_IO::write_rho` (`read_rho`) as `Module_IO::write_cube` (`read_cube`), since they are generic functions now.
2. Move `ElecStaticPot` to `ElecStaticPot.cube`, and remove `ElecStaticPot_AVE`.
3. <Test> Update `201_NO_15_f_pseudopots`, in which `out_pot=2`. Modify the `tools/catch_properties.sh`.
4. <Doc> Update the doc about `out_pot`, including the problem mentioned in https://github.com/deepmodeling/abacus-develop/issues/2166.

sunliang98 committed:
<Refactor> Unify the format of charge density, effective potential and electrostatic potential into cube.
Remove the ElecStaticPot_AVE.
rename write_rho (read_rho) as write_cube (read_cube), since they are generic functions now.

sunliang98 committed:
<Test> Update 201_NO_15_f_pseudopots, in which out_pot=2.

sunliang98 committed:
<Doc> Update the doc about out_pot.

sunliang98 committed:
Merge branch 'develop' into refactor

YuLiu98 commented:
> 2\. Move `ElecStaticPot` to `ElecStaticPot.cube`, and remove `ElecStaticPot_AVE`.

`ElecStaticPot_AVE` is useful in slab calculations. If you want to remove it, I think there should be a postprocessing tool to obtain the average ESP along the vacuum direction. Maybe the postprocessing tool could be implemented here: https://github.com/deepmodeling/abacus-develop/tree/develop/tools/plot-tools.

sunliang98 commented:
OK, I will submit a python script to do it.

sunliang98 committed:
<Tools> Add a python script `aveElecStatPot.py`.
Add a Python script located at `tools/average_pot/aveElecStatPot.py` that calculates the average electrostatic potential along the z-axis and outputs it to `ElecStaticPot_AVE`.
And update the doc about out_pot.

sunliang98 committed:
Merge branch 'refactor' of https://github.com/sunliang98/abacus-develop into refactor

sunliang98 commented:
> OK, I will submit a python script to do it.

It's done. @YuLiu98

dyzheng commented on docs/advanced/input_files/input-main.md:
```diff
@@ -2112,8 +2123,7 @@ These variables are used to control berry phase and wannier90 interface paramete
 - **Description**:
-  phase of Gauss type electric field  
-  amp*cos(2pi*f(t-t0)+phase)exp(-(t-t0)^2/2sigma^2)
+  phase of Gauss type electric fieldamp*cos(2pi*f(t-t0)+phase)exp(-(t-t0)^2/2sigma^2)
```
？

dyzheng replied:
Please explain each variable in the function

sunliang98 replied:
The format is corrected, but I'm not aware of the meaning of these variables, since I'm not familiar with TDDFT.

dyzheng commented on docs/advanced/input_files/input-main.md:
```diff
@@ -2128,8 +2138,7 @@ These variables are used to control berry phase and wannier90 interface paramete
 - **Description**:
-  step number of time center of Gauss type electric field  
-  amp*cos(2pi*f(t-t0)+phase)exp(-(t-t0)^2/2sigma^2)
+  step number of time center of Gauss type electric fieldamp*cos(2pi*f(t-t0)+phase)exp(-(t-t0)^2/2sigma^2)
```
Please explain each variable in the function

sunliang98 replied:
The format is corrected.

dyzheng commented on docs/advanced/input_files/input-main.md:
```diff
@@ -2155,8 +2164,7 @@ These variables are used to control berry phase and wannier90 interface paramete
 - **Description**:
-  phase of Trapezoid type electric field  
-  E = amp*cos(2pi*f*t+phase) t/t1 , t<t1
+  phase of Trapezoid type electric fieldE = amp*cos(2pi*f*t+phase) t/t1 , t<t1
```
Format is wrong.

sunliang98 replied:
The format is corrected.

dyzheng reviewed (changes requested):
Please correct the format in documentation

sunliang98 commented:
> Please correct the format in documentation

It's strange... I have not modified them. I will correct them.

hongriTianqi commented on source/module_io/istate_envelope.cpp:
```diff
@@ -4,7 +4,7 @@
 #include "module_base/global_variable.h"
 #include "module_io/write_wfc_pw.h"
 #include "module_io/write_wfc_r.h"
-#include "module_io/rho_io.h"
```
I suggest not to remove rho_io.h, but rather add another cube_io.h, and functions in rho_io call functions in cube_io

hongriTianqi replied:
In this way, the codes to create or open files will be able to be written inside write_rho and read_rho, respectively.

sunliang98 replied:
OK

sunliang98 replied:
It's done.

sunliang98 committed:
<Refactor> Restore the rho_io.h, write_rho.cpp, and read_rho.cpp.

sunliang98 committed:
<Doc> Correct the format in input-main.md

sunliang98 commented:
> Please correct the format in documentation

It's done.

sunliang98 committed:
Merge branch 'develop' into refactor
